src/uwp/Microsoft.Build.Net.CoreRuntimeTask/ResourceHandlingTask.cs

```diff
@@ -144,55 +174,81 @@ public override bool Execute()
                         {
-                            if (containsResources)
-                                unprocessedAssemblyList.Add(assemblyFilePath);
-                            
-                            if (unprocessedAssemblyList.Count == 0)
-                                firstNonFrameworkAssembly = assemblyFilePath;
+                            resWInfoList = ExtractAssemblyResWList(assemblyFilePath.ItemSpec, out containsFrameworkResources);
+
+                            if (resWInfoList != null)
+                            {
+                                FileInfo fiAssembly = new FileInfo(assemblyFilePath.ItemSpec);
+                                _state.Save(assemblyFilePath.ItemSpec, fiAssembly.LastWriteTimeUtc, containsFrameworkResources, resWInfoList);
+                            }
                         }
-                        else
+
+                        if (resWInfoList != null)
                         {
-                            TaskItem newTaskItem = new TaskItem(reswPath);
-                            newTaskItem.SetMetadata("NeutralResourceLanguage","en-US");
-                            newTaskItem.SetMetadata("ResourceIndexName",Path.GetFileNameWithoutExtension(reswPath));
-                            reswList.Add(newTaskItem);
+                            foreach (ResWInfo reswInfo in resWInfoList)
+                            {
+                                TaskItem newTaskItem = new TaskItem(reswInfo.ResWPath);
+                                newTaskItem.SetMetadata("ResourceIndexName", reswInfo.ResourceIndexName);
+                                if (!String.IsNullOrEmpty(reswInfo.NeutralResourceLanguage))
+                                {
+                                    newTaskItem.SetMetadata("NeutralResourceLanguage", reswInfo.NeutralResourceLanguage);
+                                }
+
+                                if (containsFrameworkResources)
+                                {
+                                    frameworkReswList.Add(newTaskItem);
+                                }
+                                else
+                                {
+                                    newTaskItem.SetMetadata("OriginalItemSpec", reswInfo.ResWPath); // Original GenerateResource behavior creates this additional metadata item on processed non-framework assemblies
+                                    reswList.Add(newTaskItem);
+                                }
+                            }
                         }
 
                     }
 
-                    UnprocessedAssemblyList = unprocessedAssemblyList.ToArray();
+                    UnprocessedAssemblyList = unprocessedAssemblyList.ToArray(); // For now this list will always be empty
                     
-                    if (!SkipFrameworkResources)
+                    if (SkipFrameworkResources)
```

Comment:
>SkipFrameworkResources [](start = 24, length = 22)

would it be better passing SkipFrameworkResources instead of containsFrameworkResources through the code and avoid using 2 lists for the framework and non framework? maybe this can simplify the code?

Reply:
We will always need two different Booleans. `SkipFrameworkResources` is an input and `containsFrameworkResources` will tell us how to process an input Framework assembly.

I've made a code change to address this.